hibernate-core/src/test/java/org/hibernate/test/component/empty/EmptyCompositeIdAttributeTest.java

```diff
@@ -0,0 +1,186 @@
+		configuration.getProperties().put( Environment.USE_SECOND_LEVEL_CACHE, "false" );
+	}
+
+	@Test
+	@TestForIssue( jiraKey = "HHH-11898" )
+	public void testPersistNullAttributeEmptyDisabled() {
+		testPersistNullAttribute( false );
+	}
+
+	@Test
+	@TestForIssue( jiraKey = "HHH-11898" )
+	public void testPersistNullAttributeEmptyEnabled() {
+		testPersistNullAttribute( true );
+	}
+
+	private void testPersistNullAttribute(boolean isEmptyEnabled) {
+		rebuildSessionFactory( c -> c.setProperty(
+									   AvailableSettings.CREATE_EMPTY_COMPOSITES_ENABLED,
+									   String.valueOf( isEmptyEnabled )
+							   )
+		);
+
+		int entityAId = doInHibernate(
+				this::sessionFactory,
+				session -> {
+					final EntityB entityB = new EntityB();
+					entityB.bpk = new BPK();
+					// explicitly set the default int values to make clear
+					// that these are "real" values
+					entityB.bpk.intVal1 = 0;
+					entityB.bpk.intVal2 = 0;
+					session.persist( entityB );
+
+					final EntityA entityA = new EntityA();
+					// entityA.selectedEntityBPK is null
+					session.persist( entityA );
+					return entityA.id;
+				}
+		);
+
+		doInHibernate(
+				this::sessionFactory,
+				session -> {
+					final EntityA entityA = session.get( EntityA.class, entityAId );
+					if ( isEmptyEnabled ) {
+						// entityA.selectedEntityBPK should be an empty composite
+						assertNotNull( entityA.selectedEntityBPK );
+						assertNull( session.get( EntityB.class, entityA.selectedEntityBPK ) );
+					}
+					else {
+						assertNull( entityA.selectedEntityBPK );
+					}
+				}
+		);
+	}
+
+	@Test
+	@TestForIssue( jiraKey = "HHH-11898" )
+	public void testUpdateAttributeEmptyDisabled() {
+		testUpdateAttribute( false );
+	}
+
+	@Test
+	@TestForIssue( jiraKey = "HHH-11898" )
+	public void testUpdateAttributeEmptyEnabled() {
+		testUpdateAttribute( true );
+	}
+
+	private void testUpdateAttribute(boolean isEmptyEnabled) {
+		rebuildSessionFactory( c -> c.setProperty(
+									   AvailableSettings.CREATE_EMPTY_COMPOSITES_ENABLED,
+									   String.valueOf( isEmptyEnabled )
+							   )
+		);
+
+		int entityAId = doInHibernate(
+				this::sessionFactory,
+				session -> {
+					final EntityB entityB = new EntityB();
+					entityB.bpk = new BPK();
+					// explicitly set the default int values to make clear
+					// that these are "real" values
+					entityB.bpk.intVal1 = 0;
+					entityB.bpk.intVal2 = 0;
+					session.persist( entityB );
+
+					final EntityA entityA = new EntityA();
+					// entityA.selectedEntityBPK is null
+					session.persist( entityA );
+					return entityA.id;
+				}
+		);
+
+		doInHibernate(
+				this::sessionFactory,
+				session -> {
+					final EntityA entityA = session.get( EntityA.class, entityAId );
+					entityA.selectedEntityBPK = new BPK();
+					// explicitly set the default int values to make clear
+					// that these are "real" values
+					entityA.selectedEntityBPK.intVal1 = 0;
+					entityA.selectedEntityBPK.intVal2 = 0;
```

Comment:
Suppose HHH-11898 is fixed so that an empty composite with primitive values are equivalent to `null` if primitive attributes are set to the primitive default and non-primitive attributes are null. 

When the session is flushed, `entityA` will *not* be updated because `entityA.selectedEntityBPK` will be equivalent to null. 

This will happen even with `hibernate.create_empty_composites.enabled=false`!